kurlkinds/pkg/lint/rego/vars.rego

```diff
@@ -0,0 +1,585 @@
+package kurl.installer
+
+# known_versions holds all addons and their respective versions, latest
+# is kept on its own variable.
+default known_versions = {
+	"antrea": {
+		"latest": "1.4.0",
+		"versions": [
+			"1.4.0",
+			"1.2.1",
+			"1.2.0",
+			"1.1.0",
+			"1.0.1",
+			"1.0.0",
+			"0.13.1"
+		]
+	},
+	"aws": {
+		"latest": "0.1.0",
+		"versions": [
+			"0.1.0"
+		]
+	},
+	"calico": {
+		"latest": "3.9.1",
+		"versions": [
+			"3.9.1"
+		]
+	},
+	"certManager": {
+		"latest": "1.9.1",
+		"versions": [
+			"1.9.1",
+			"1.0.3"
+		]
+	},
+	"collectd": {
+		"latest": "v5",
+		"versions": [
+			"v5",
+			"0.0.1"
+		]
+	},
+	"containerd": {
+		"latest": "1.6.8",
+		"versions": [
+			"1.6.8",
+			"1.6.7",
+			"1.6.6",
+			"1.6.4",
+			"1.5.11",
+			"1.5.10",
+			"1.4.13",
+			"1.4.12",
+			"1.4.11",
+			"1.4.10",
+			"1.4.9",
+			"1.4.8",
+			"1.4.6",
+			"1.4.4",
+			"1.4.3",
+			"1.3.9",
+			"1.3.7",
+			"1.2.13"
+		]
+	},
+	"contour": {
+		"latest": "1.22.1",
+		"versions": [
+			"1.22.1",
+			"1.22.0",
+			"1.21.1",
+			"1.21.0",
+			"1.20.1",
+			"1.20.0",
+			"1.19.1",
+			"1.18.0",
+			"1.16.0",
+			"1.15.1",
+			"1.14.1",
+			"1.14.0",
+			"1.13.1",
+			"1.13.0",
+			"1.12.0",
+			"1.11.0",
+			"1.10.1",
+			"1.7.0",
+			"1.0.1",
+			"0.14.0"
+		]
+	},
+	"docker": {
+		"latest": "20.10.17",
+		"versions": [
+			"20.10.17",
+			"20.10.5",
+			"19.03.15",
+			"19.03.10",
+			"19.03.4",
+			"18.09.8"
+		]
+	},
+	"ekco": {
+		"latest": "0.22.0",
+		"versions": [
+			"0.22.0",
+			"0.21.1",
+			"0.21.0",
+			"0.20.0",
+			"0.19.9",
+			"0.19.6",
+			"0.19.3",
+			"0.19.2",
+			"0.19.1",
+			"0.19.0",
+			"0.18.0",
+			"0.17.0",
+			"0.16.0",
+			"0.15.0",
+			"0.14.0",
+			"0.13.0",
+			"0.12.0",
+			"0.11.0",
+			"0.10.3",
+			"0.10.2",
+			"0.10.1",
+			"0.10.0",
+			"0.9.0",
+			"0.8.0",
+			"0.7.0",
+			"0.6.0",
+			"0.5.0",
+			"0.4.2",
+			"0.4.1",
+			"0.4.0",
+			"0.3.0",
+			"0.2.4",
+			"0.2.3",
+			"0.2.1",
+			"0.2.0",
+			"0.1.0"
+		]
+	},
+	"fluentd": {
+		"latest": "1.7.4",
+		"versions": [
+			"1.7.4"
+		]
+	},
+	"goldpinger": {
+		"latest": "3.6.1-5.4.2",
+		"versions": [
+			"3.6.1-5.4.2",
+			"3.5.1-5.2.0",
+			"3.3.0-5.1.0",
+			"3.2.0-5.0.0",
+			"3.2.0-4.2.1",
+			"3.2.0-4.1.1"
+		]
+	},
+	"k3s": {
+		"latest": "v1.23.3+k3s1",
+		"versions": [
+			"v1.23.3+k3s1",
+			"v1.19.7+k3s1"
+		]
+	},
+	"kotsadm": {
+		"latest": "1.86.2",
+		"versions": [
```

Comment:
how will this work with kots as an external add-on?

Reply:
What I have done so far is: if the add-on version is invalid but it has an s3override property set then it is considered to be valid.